CrossPlatform/BasePage.cs

```diff
@@ -14,7 +14,9 @@ public abstract class BasePage
         protected bool OnAndroid { get; private set; }
         protected bool OniOS { get; private set; }
 
-        protected Func<AppQuery, AppQuery> Trait { get; set; }
+        protected abstract Trait Trait { get; }
+
+        protected Func<AppQuery, AppQuery> CurrentTrait { get; private set; }
```

Comment:
Yeah I definitely see the dilemma. How about offloading this logic to the trait directly? Make `AndroidTrait` `iOSTrait` set only and then have a third get only `Current` property. I think it makes sense for the trait to know which one to use.